src/advisories/cfg.rs

```diff
@@ -303,21 +305,29 @@ impl<'de> Deserialize<'de> for Config {
         let maximum_db_staleness = maximum_db_staleness
             .unwrap_or_else(|| Spanned::new(Duration::seconds_f64(NINETY_DAYS)));
 
+        let deprecated = if version <= 1 {
+            Some(Deprecated {
+                vulnerability: vulnerability.unwrap_or(LintLevel::Deny),
+                unmaintained: unmaintained.unwrap_or(LintLevel::Warn),
+                unsound: unsound.unwrap_or(LintLevel::Warn),
+                notice: notice.unwrap_or(LintLevel::Warn),
+                severity_threshold,
+            })
+        } else {
+            None
+        };
```

Comment:
Since you're skipping the deprecated fields here if `version >= 2`, would it be more sensible to return an error when the fields are still set?